fuzz/src/peer_crypt.rs

```diff
@@ -1,78 +1,346 @@
-					return;
-				}
-				read_pos += slice_len;
-				&data[read_pos - slice_len..read_pos]
+impl<'a> FuzzGen<'a> {
+	fn new(data: &'a [u8]) -> Self {
+		Self {
+			read_pos: 0,
+			data
+		}
+	}
+
+	fn generate_bytes(&mut self,  num: usize) -> Result<&'a [u8], GeneratorFinishedError> {
+		if self.data.len() < self.read_pos + num {
+			return Err(GeneratorFinishedError { });
+		}
+
+		self.read_pos += num;
+		Ok(&self.data[self.read_pos - num..self.read_pos])
+	}
+
+	fn generate_secret_key(&mut self) -> Result<SecretKey, GeneratorFinishedError> {
+		// Loop through the input 32 bytes at a time until a valid
+		// secret key can be created or we run out of bytes
+		loop {
+			match SecretKey::from_slice(self.generate_bytes(32)?) {
+				Ok(key) => { return Ok(key) },
+				_ => { }
 			}
 		}
 	}
 
-	let our_network_key = match SecretKey::from_slice(get_slice!(32)) {
-		Ok(key) => key,
-		Err(_) => return,
-	};
-	let ephemeral_key = match SecretKey::from_slice(get_slice!(32)) {
-		Ok(key) => key,
-		Err(_) => return,
-	};
+	fn generate_bool(&mut self) -> Result<bool, GeneratorFinishedError> {
+		let next_byte = self.generate_bytes(1)?;
+		Ok(next_byte[0] & 1 == 1)
+	}
+}
+
+#[derive(PartialEq)]
+struct GeneratorFinishedError { }
 
-	let mut crypter = if get_slice!(1)[0] != 0 {
-		let their_pubkey = match PublicKey::from_slice(get_slice!(33)) {
-			Ok(key) => key,
-			Err(_) => return,
+impl fmt::Debug for GeneratorFinishedError {
+	fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+		formatter.write_str("Generator out of bytes")
+	}
+}
+impl fmt::Display for GeneratorFinishedError {
+	fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+		formatter.write_str("Generator out of bytes")
+	}
+}
+impl error::Error for GeneratorFinishedError {
+	fn description(&self) -> &str { "Generator out of bytes" }
+}
+
+struct TestCtx {
+	initiator_static_public_key: PublicKey,
+	initiator_handshake: PeerHandshake,
+	responder_static_public_key: PublicKey,
+	responder_handshake: PeerHandshake,
+	act1: Vec<u8>
+}
+
+impl TestCtx {
+	fn make(generator: &mut FuzzGen) -> Result<Self, GeneratorFinishedError> {
+		let curve = secp256k1::Secp256k1::new();
+
+		// Generate needed keys for successful handshake
+		let initiator_static_private_key = generator.generate_secret_key()?;
+		let initiator_static_public_key = PublicKey::from_secret_key(&curve, &initiator_static_private_key);
+		let initiator_ephemeral_private_key = generator.generate_secret_key()?;
+		let responder_static_private_key = generator.generate_secret_key()?;
+		let responder_static_public_key = PublicKey::from_secret_key(&curve, &responder_static_private_key);
+		let responder_ephemeral_private_key = generator.generate_secret_key()?;
+
+		let mut initiator_handshake = PeerHandshake::new_outbound(&initiator_static_private_key, &responder_static_public_key, &initiator_ephemeral_private_key);
+		let act1 = initiator_handshake.set_up_outbound();
```

Comment:
squashed with `review: Non-controversial renames, nits, small fixups`

I've included these as raw values since making the constant values in peers/handshake public is a bit of a mess right now. Fuzzing is a separate test environment so I think this is OK especially considering this code is unlikely to change.